src/mscorlib/src/System/Lazy.cs

```diff
@@ -322,147 +516,12 @@ public T Value
         {
             get
             {
-                Boxed boxed = null;
-                if (m_boxed != null )
-                {
-                    // Do a quick check up front for the fast path.
-                    boxed = m_boxed as Boxed;
-                    if (boxed != null)
-                    {
-                        return boxed.m_value;
-                    }
-
-                    LazyInternalExceptionHolder exc = m_boxed as LazyInternalExceptionHolder;
-                    Debug.Assert(exc != null);
-                    exc.m_edi.Throw();
-                }
-
-                // Fall through to the slow path.
-                return LazyInitValue();
+                var state = _state;
+                if (state == null)
+                    return _value;
+                return LazyGetValue(state);
```

Comment:
Actually I think I'm worried about nothing. This is what thinking on public transport does to you! Anyway I'll still contemplate it properly when I have time!